### PR TITLE
feat: Copy section address when clicking section anchor icons

### PR DESCRIPTION
**Pull request by gabriele-ct**
Fixes: #699

### CODE REVIEW

File: packages/ui-kit/src/components/markdown.tsx [hunk 1]
```diff
@@ -298,36 +300,64 @@ const TypographyContainer = styled.div`
...
+const withCopyToClipboard =
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  (Component: React.ComponentType) => (props: any) => {
+    const [isCopiedToClipboard, setIsCopiedToClipboard] = React.useState(false);
+    const handleCopyToClipboardClick = (event: React.SyntheticEvent) => {
+      event.preventDefault();
+      const sectionUrl = `${window.location.href.split('#')[0]}#${props.id}`;
+      copyToClipboard(sectionUrl);
+      setIsCopiedToClipboard(true);
+      setTimeout(() => setIsCopiedToClipboard(false), 1500);
```
Comment (nkuehn) [inline packages/ui-kit/src/components/markdown.tsx:326]: Nitpick: the timeout based implementation works but has am edge case when you keep the pointer hovering over the item longer - the text changes back then, which seems erratic. 

Did you try using the `onClose` callback of the tooltip UI-Kit component to reset the state?  It's not in the "knobs" UI of the UI-Kit storybook but in the readme. 

https://uikit.commercetools.com/?path=/story/components-tooltips--tooltip

File: packages/ui-kit/src/components/markdown.tsx [hunk 2]
```diff
@@ -298,36 +300,64 @@ const TypographyContainer = styled.div`
...
-  );
-};
+          > * + * {
+            margin: 0 0 0 ${dimensions.spacings.s};
+          }
+        `}
+      >
+        <span>{props.children}</span>
+        <Tooltip
+          title={isCopiedToClipboard ? 'Copied' : 'Copy to clipboard'}
```
Comment (nkuehn) [inline packages/ui-kit/src/components/markdown.tsx:350]:
```suggestion
          title={isCopiedToClipboard ? 'Copied' : 'Copy link to clipboard'}
```
just tuning the language

File: packages/ui-kit/src/components/markdown.tsx (hunk 1 above)
Comment (gabriele-ct) [reply packages/ui-kit/src/components/markdown.tsx:326]: @nkuehn works like a charm! Thank you, I guess we could implement something similar for the `code-block` component which still uses the timeout based approach: [`packages/ui-kit/src/components/code-block.tsx](https://github.com/commercetools/commercetools-docs-kit/blob/65b2345e6004dc7b69dbe76e3ede794ca5dc0906/packages/ui-kit/src/components/code-block.tsx#L213)
I'll create a ticket for that

**Review by timonrey — APPROVED** [review]
Thanks! Looks great! 🚀 🙌🏼

**Pull request merged**